js/ui/gantt/ui.gantt.js

```diff
@@ -0,0 +1,152 @@
+            onRowCollapsed: () => this._updateGanttView(),
+            onRowExpanded: () => this._updateGanttView()
+        });
+    }
+
+    _initSplitter() {
+        $('<div>')
+            .addClass(GANTT_SPLITTER_CLASS)
+            .appendTo(this.$element());
+    }
+
+    _initGanttView() {
+        if(this._ganttView) {
+            return;
+        }
+        const $ganttView = $('<div>')
+            .addClass(GANTT_VIEW_CLASS)
+            .appendTo(this.$element());
+
+        this._ganttView = this._createComponent($ganttView, GanttView, {
+            height: this._treeList._$element.get(0).offsetHeight,
+            tasks: this._getTasks(),
+            dependenciesDataSource: this.option('dependenciesDataSource'),
+            resourcesDataSource: this.option('resourcesDataSource'),
+            resourceAssignmentsDataSource: this.option('resourceAssignmentsDataSource'),
+            onSelectionChanged: (e) => { this._onGanttViewSelectionChanged(e); },
+            onScroll: (e) => { this._onGanttViewScroll(e); }
+        });
+    }
+
+    _onTreeListContentReady(e) {
+        if(e.component.getDataSource()) {
+            this._initGanttView();
+            this._initScrollSync(e.component);
+        }
+    }
+    _onGanttViewSelectionChanged(e) {
+        this._treeList.option('selectedRowKeys', [e.id]);
+    }
+    _onGanttViewScroll(e) {
+        const treeListScrollable = this._treeList.getScrollable();
+        if(treeListScrollable) {
+            const diff = e.scrollableElement.scrollTop - treeListScrollable.scrollTop();
+            if(diff !== 0) {
+                treeListScrollable.scrollBy(diff);
+            }
+        }
+    }
+    _onTreeListScroll(e) {
+        const ganttViewScrollable = this._ganttView._getScrollable();
+        if(ganttViewScrollable.scrollTop !== e.scrollOffset.top) {
+            ganttViewScrollable.scrollTop = e.scrollOffset.top;
+        }
+    }
+
+    _updateGanttView() {
+        this._ganttView.option('tasks', this._getTasks());
+        this._ganttView._update();
+    }
+    _getTasks() {
+        return this._treeList.getVisibleRows().map(r => r.data);
+    }
+    _initScrollSync(treeList) {
+        const scrollable = treeList.getScrollable();
+        if(scrollable) {
+            scrollable.off('scroll', (e) => { this._onTreeListScroll(e); });
+            scrollable.on('scroll', (e) => { this._onTreeListScroll(e); });
+        }
+    }
+
+    _getDefaultOptions() {
+        return extend(super._getDefaultOptions(), {
+            /**
+            * @name dxGanttOptions.taskDataSource
+            * @type object
+            * @default null
+            */
+            taskDataSource: null,
+            /**
+            * @name dxGanttOptions.dependenciesDataSource
+            * @type object
+            * @default null
+            */
+            dependenciesDataSource: null,
+            /**
+            * @name dxGanttOptions.resourcesDataSource
```

Comment:
Do we need that "dataSource" postfix in every option name? IMHO, simple names like `tasks`, `dependencies`, `resources`, and `resourceAssignments` (or `resourceAssignees`, see my comment below) are much better